pkg/reconciler/inmemorychannel/dispatcher/controller.go

```diff
@@ -137,6 +140,13 @@ func NewController(
 				DeleteFunc: r.deleteFunc,
 			}})
 
+	inmemorychannelInformer.Informer().AddEventHandler(controller.HandleAll(
+		controller.EnsureTypeMeta(
+			r.tracker.OnChanged,
+			messagingv1.SchemeGroupVersion.WithKind("InMemoryChannel"),
+		),
+	))
```

Comment:
It's not the tracker the solution but `EnsureTypeMeta` therefore where we set the handler (just above) we need to call `EnsureTypeMeta`

```golang
gvk := messagingv1.SchemeGroupVersion.WithKind("InMemoryChannel")


				AddFunc:    controller.EnsureTypeMeta(impl.Enqueue, gvk)
				UpdateFunc: controller.PassNew(controller.EnsureTypeMeta(impl.Enqueue, gvk)),
				DeleteFunc: controller.EnsureTypeMeta(r.deleteFunc, gvk),
```

that ensures that whatever object is passed to callback has the `Typemeta` based on `gvk`

Reply:
Okay thanks that makes sense!